src/user-subscription.js

```diff
@@ -70,53 +64,24 @@ export class UserSubscription {
   onAddedToRoom = ({ room: roomData }) => {
     const basicRoom = parseBasicRoom(roomData)
     this.roomStore.set(basicRoom.id, basicRoom).then(room => {
-      if (this.hooks.global.onAddedToRoom) {
-        this.hooks.global.onAddedToRoom(room)
-      }
+      this.hooks.internal.onAddedToRoom(basicRoom.id).then(() => {
+        if (this.hooks.global.onAddedToRoom) {
+          this.hooks.global.onAddedToRoom(room)
+        }
+      })
     })
   }
 
   onRemovedFromRoom = ({ room_id: roomId }) => {
     this.roomStore.pop(roomId).then(room => {
       // room will be undefined if we left with leaveRoom
+      this.hooks.internal.onRemovedFromRoom(roomId)
```

Comment:
Why do we expect a promise from `this.hooks.internal.onAddedToRoom` (and wait for it), but not for `this.hooks.internal.onRemovedFromRoom`?